### PR TITLE
Confirmation window

### PR DESCRIPTION
**Pull request by akhmorgan**

### CODE REVIEW

**Comment by WPFerg** [comment]
A few things from using (probably good to run them past Rui)
* The cancel button doesn't look too great (Perhaps have the OK button's style on cancel and have OK with inverted BG/text?)
* Should there be behaviour on pressing Enter/Tab?
* Should the confirm box go away when clicking anywhere on the box that isn't OK/Cancel/X?

File: src/sidebars/star/star.html [hunk 1]
```diff
@@ -1,2 +1,20 @@
 ﻿<img class="star" ng-src="assets/png/{{starCtrl.favouriteUrl()}}.png" ng-click="starCtrl.click(); $event.stopPropagation();"
      ng-mouseenter="starCtrl.mouseEnter()" ng-mouseleave="starCtrl.mouseLeave()" draggable="false" />
+
+<div class="confirmation-backdrop " ng-show="starCtrl.confirmationShow" ng-click="starCtrl.hideModal()">
```
Comment (WPFerg) [inline src/sidebars/star/star.html]: Maybe swap out `ng-show` for `ng-if` so all the confirm boxes aren't lingering in the DOM.

**Comment by akhmorgan** [comment]
@wpferg I agree. This all needs to be run past Rui. This is just the initial implementation to get the ball rolling.

File: src/sidebars/star/star.html (hunk 1 above)
Comment (akhmorgan) [inline src/sidebars/star/star.html]: Good point. I'll try that.

**Comment by owennw** [comment]
Something in here affects the styling of the search tab in main view. The search results appear too high up and interfere with the search input box.

File: src/sidebars/star/star.less [hunk 2]
```diff
@@ -1,3 +1,49 @@
...
+    margin: 100px auto;
+    .modal-content {
+      .header{
+        padding: 15px;
+        border-bottom: 1px solid #e5e5e5;
+        .title {
+          padding-top: 1px;
+          font-size: 20px;
+          font-weight: 500;
+          line-height: 1;
```
Comment (owennw) [inline src/sidebars/star/star.less]: There's a mixin `.font` for the size, weight and line height.

File: src/sidebars/star/star.less [hunk 3]
```diff
@@ -1,3 +1,49 @@
...
+
+.confirmation-backdrop {
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  z-index: 1050;
+  display: none;
+  overflow: hidden;
```
Comment (owennw) [inline src/sidebars/star/star.less]: I think there's a mixin for overflow too...

File: src/sidebars/star/star-controller.js [hunk 4]
```diff
@@ -47,6 +52,19 @@
...
         }
 
+        deselect() {
+            if (!this.store) {
+                this.store = window.storeService.open(window.name);
+            }
+            this.stock.favourite = false;
+            this.store.remove(this.stock);
+            this.hideModal();
+        }
```
Comment (owennw) [inline src/sidebars/star/star-controller.js]: You can probably call `deselect` from `click` too?

File: src/sidebars/star/star-controller.js [hunk 5]
```diff
@@ -32,7 +34,10 @@
         }
 
         click() {
-            if (!this.check || confirm('Are you sure you wish to remove this stock (' + this.stock.code + ') from your favourites?')) {
+            if (this.check) {
+                this.confirmationShow = true;
+            }
+            if (!this.check) {
```
Comment (owennw) [inline src/sidebars/star/star-controller.js]: This can be an `else`.

File: src/sidebars/star/star-controller.js (hunk 4 above)
Comment (akhmorgan) [inline src/sidebars/star/star-controller.js]: Done.

File: src/sidebars/star/star.less (hunk 2 above)
Comment (akhmorgan) [inline src/sidebars/star/star.less]: Done.

File: src/sidebars/star/star.less (hunk 3 above)
Comment (akhmorgan) [inline src/sidebars/star/star.less]: And done.

File: src/sidebars/star/star.html [hunk 6]
```diff
@@ -1,2 +1,20 @@
 ﻿<img class="star" ng-src="assets/png/{{starCtrl.favouriteUrl()}}.png" ng-click="starCtrl.click(); $event.stopPropagation();"
      ng-mouseenter="starCtrl.mouseEnter()" ng-mouseleave="starCtrl.mouseLeave()" draggable="false" />
+
+<div class="confirmation-backdrop " ng-if="starCtrl.confirmationShow" ng-click="starCtrl.hideModal()">
+  <div class="confirmation-modal">
```
Comment (WPFerg) [inline src/sidebars/star/star.html]: Clicking on the modal itself closes it. Would a `$event.stopPropagation()` here fix this?

File: src/sidebars/star/star.html [hunk 7]
```diff
@@ -1,2 +1,20 @@
...
+      <div class="header">
+        <span class="title">StockFlux</span>
+        <icon name="close" icon-click="starCtrl.hideModal()"></icon>
+      </div>
+      <div class="message">
+        <p ng-bind="starCtrl.confirmationMessage"></p>
+      </div>
+      <div class="buttons">
+        <button class="btn btn-primary ng-binding" ng-click="starCtrl.deselect(); $event.stopPropagation();">OK</button>
+        <button class="btn btn-default ng-binding" ng-click="starCtrl.hideModal(); $event.stopPropagation();">Cancel</button>
```
Comment (owennw) [inline src/sidebars/star/star.html]: I thought `btn` and `btn-primary` etc were bootstrap classes, but we're not using bootstrap. Are they doing something here?

**Comment by bjedrzejewski** [comment]
Clicking on the window itself shouldn't close it.

**Comment by bjedrzejewski** [comment]
@akhmorgan @owennw do you think it would be better to always show the dialog in the middle of the window?

File: src/sidebars/star/star.html (hunk 7 above)
Comment (bjedrzejewski) [inline src/sidebars/star/star.html]: I think we are inheriting these bootstrap customized classes from d3fc/bootstrap. It is not such a bad idea I think! We are going to be the same as the showcase.

**Comment by akhmorgan** [comment]
@bjedrzejewski I think that sort of thing is more a question for UX, which can then be followed up in another PR. This PR is just to get the core functionality in.

**Comment by bjedrzejewski** [comment]
@akhmorgan Ok, I would say then- I am happy to merge if one more person is happy to merge. @owennw - looking at you

**Comment by owennw** [comment]
The spec states that the dialog should appear close to the star icon, with the bubble like the closed window list.

**Pull request closed without merging**